docroot/sites/research.uiowa.edu/modules/research_migrate/src/Plugin/migrate/source/Articles.php

```diff
@@ -0,0 +1,277 @@
+        '@old' => $row->getSourceProperty('nid'),
+        '@article' => $row->getSourceProperty('title'),
+      ]);
+    }
+
+    // Set source link and link directly. Should be one or the other, not both.
+    $external_link = $row->getSourceProperty('field_ovpred_article_ext_url');
+    $now_link = $row->getSourceProperty('field_ovpred_article_ian_url');
+    if (!empty($external_link)) {
+      $row->setSourceProperty('custom_source_link', $external_link[0]['url']);
+    }
+    elseif (!empty($now_link)) {
+      $row->setSourceProperty('custom_source_link', $now_link[0]['url']);
+    }
+
+    $article_type = $row->getSourceProperty('field_ovpred_article_type')[0]['value'];
+    if ($article_type === 'external' || $article_type === 'ianow') {
+      $row->setSourceProperty('field_article_source_link_direct', 1);
+    }
+
+    if ($image = $row->getSourceProperty('field_ovpred_article_image')) {
+      $row->setSourceProperty('field_image', $this->processImageField($image[0]['fid'], $image[0]['alt'], $image[0]['title']));
+    }
+
+    // Establish an array to eventually map to field_contact_reference.
+    $nids = [];
+
+    // Set our contactMap of person nodes if it's not already.
+    if (empty($this->contactMap)) {
+      $this->contactMap = \Drupal::database()
+        ->select('node_field_data', 'n')
+        ->fields('n', ['title', 'nid'])
+        ->condition('n.type', 'person', '=')
+        ->execute()
+        ->fetchAllKeyed();
+    }
+
+    // Migrate article contacts from old site
+    // and do some re-mapping in the process.
+    $contact_nids = $row->getSourceProperty('field_ovpred_article_contact_target_id');
+    if (!empty($contact_nids)) {
+      // Fetch node data using NIDs from our old site.
+      $contact_results = $this->select('node', 'n')
+        ->fields('n', ['title', 'nid'])
+        ->condition('n.nid', $contact_nids, 'IN')
+        ->execute();
+
+      foreach ($contact_results as $result) {
+        // Check contactMapping and create node.
+        $mapping_result = $this->contactMapping($result['nid']);
+        if ($mapping_result != NULL) {
+          if ($mapping_result != '0') {
+            // There is a mapping result, so switch to use that target_id.
+            $nids[] = $mapping_result;
+          }
+          else {
+            // This contact should be skipped and not attached to the content.
+            break;
+          }
+        }
+        else {
+          // Create person nodes if not mapped but are POCs on source.
+          $explode_name = explode(' ', $result['title']);
+          $node = Node::create([
+            'title' => $result['title'],
+            'field_person_first_name' => $explode_name[0],
+            'field_person_last_name' => $explode_name[1],
+            'type' => 'person',
+            'status' => 0,
+          ]);
+          if ($node->save()) {
+            $nids[] = $node->id();
+          }
+        }
```

Comment:
I think if there's a person listed as a contact on the source, but not included in the mapping provided to us, they should be skipped.

@briand44 The issue says that we are not creating people as part of this, but was it specifically so that we don't add new people or because that was an extra complication that wasn't important?

Reply:
My understanding was that they have already created all of the contacts as people so there shouldn't be a need to create a new person. If we happen to run into a case where a person doesn't exist, I would be fine with skipping because we don't know if that person should still be the contact or not (possibly left the University)

Reply:
```suggestion

```

Reply:
That makes sense. Missed that detail.